collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/DeleteSavedFormTest.kt

```diff
@@ -44,15 +45,35 @@ class DeleteSavedFormTest {
     }
 
     @Test
-    fun whenFinalizedFormHasCreatedALocalEntity_doesNotAppearInListToDelete() {
+    fun whenFinalizedButNotSentFormHasCreatedALocalEntity_doesNotAppearInListToDelete() {
         testDependencies.server.addForm("one-question-entity-registration.xml")
 
         rule.withMatchExactlyProject(testDependencies.server.url)
+            // Drafts can be deleted
```

Comment:
I continue to be quite confident that before the change in this PR, drafts could not be deleted. Probably QA didn't think about that case because the prior PRs were framed as not allowing delete before send. In fact, we want to block delete for finalized/failed submissions and allow delete in other cases.

I think a better name for the db column would have been `canDeleteFinalized`. Maybe we should even have considered something like `mutatesEntity`. Since these db changes haven't been published yet, should we maybe consider making the change to avoid mistakes in the future?